main.js

```diff
@@ -0,0 +1,256 @@
+"use strict";
+
+const {context} = require("@arangodb/locals");
+const createRouter = require("@arangodb/foxx/router");
+const {query, db} = require("@arangodb");
+const graph_module = require("@arangodb/general-graph");
+const queues = require("@arangodb/foxx/queues");
+const joi = require("joi");
+const {modelTypes, metadataCollectionName} = require("./scripts/setup");
```

Comment:
That's probably not a great idea. You should extract these exports into a separate utility file and import them in the setup script as well. In order to provide these exports, ArangoDB will otherwise parse and execute the setup script first. Although we do run the setup script more than once anyway, this is probably a bit excessive and avoidable.

Reply:
I've extracted this to `./model/model_metadata.js`